service/service_test.go

```diff
@@ -74,130 +69,68 @@ func TestService_Recovery(t *testing.T) {
 		challengeGroups[i] = challengeGroup
 	}
 
-	submittedChallenges := make(map[int][]challenge)
-	submitChallenges := func(roundIndex int, groupIndex int) {
-		challengesGroup := challengeGroups[groupIndex]
-		for _, challenge := range challengeGroups[groupIndex] {
+	submitChallenges := func(roundID string, challenges []challenge) {
+		for _, challenge := range challenges {
 			verifier.EXPECT().Verify(gomock.Any(), challenge.data, nil).Return(&challenge_verifier.Result{Hash: challenge.data, NodeId: challenge.nodeID}, nil)
 			round, hash, err := s.Submit(context.Background(), challenge.data, nil)
 			req.NoError(err)
 			req.Equal(challenge.data, hash)
-			req.Equal(strconv.Itoa(roundIndex), round.ID)
-
-			// Verify that all submissions returned the same round instance.
-			if rounds[roundIndex] == nil {
-				rounds[roundIndex] = round
-			} else {
-				req.Equal(rounds[roundIndex], round)
-			}
+			req.Equal(roundID, round)
 		}
-
-		// Track the submitted challenges per-round for later validation.
-		submittedChallenges[roundIndex] = append(submittedChallenges[roundIndex], challengesGroup...)
 	}
 
 	// Submit challenges to open round (0).
-	submitChallenges(0, 0)
-
-	// Verify that round is still open.
-	req.Equal(rounds[0].ID, s.openRoundID())
+	submitChallenges("0", challengeGroups[0])
 
 	// Wait for round 0 to start executing.
```

Comment:
The comment says waiting for round 0 to execute, but the assert checks if round 1 is open?

Reply:
It relies on the fact that rounds IDs are sequential. Round 0 must have started execution if round 1 is open.

Reply:
It waits for round 1 to open, not round 0 to start executing. It's confusing if a comment says something different from the code that follows.